zerver/lib/actions.py

```diff
@@ -2997,3 +3005,34 @@ def make_dict(row):
     streams.sort(key=lambda elt: elt["name"])
 
     return streams
+
+def do_claim_attachment(message):
+
+    def extract_path_id(url):
+        # This check is probably not needed anymore
```

Comment:
(Might make sense to eliminate this function and just inline it below for maximum code clarity)